### PR TITLE
feat: task send transaction(s)

### PR DESCRIPTION
**Pull request by aorumbayev**
# Proposed changes

- Adding ability to send transactions from goal clerk compatible signed txn objects
- Supports inputs via --file, --transaction options OR piped input

## Extra fixes

- Adding custom json encoder for sign method when confirmation popup shows up to display canonical goal clerk compatible txn fields AND fix the bug that would crash on encoding of AppCall transactions

### CODE REVIEW

File: src/algokit/cli/tasks/send_transaction.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        file_content = json.loads(raw_file_content)
+    except json.JSONDecodeError as ex:
+        raise click.ClickException("Invalid piped transaction content!") from ex
+
+    # Check if the content is a list of dicts with the required fields
+    if not isinstance(file_content, list) or not all(_is_sign_task_output_txn(item) for item in file_content):
+        raise click.ClickException("Invalid piped transaction content!")
+
+    # Convert the content into SignedTransaction objects
+    return [encoding.msgpack_decode(item["content"]) for item in file_content]  # type: ignore[no-untyped-call]
```
Comment (robdmoore) [inline src/algokit/cli/tasks/send_transaction.py:56]: what is `"content"`?

File: src/algokit/cli/tasks/send_transaction.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    Args:
+        network (str): The network to which the transactions will be sent.
+        txns (list[SignedTransaction]): A list of signed transactions to be sent.
+
+    Returns:
+        None: The function does not return any value.
+    """
+    algod_client = load_algod_client(network)
+
+    if any(txn.transaction.group for txn in txns):
```
Comment (robdmoore) [inline src/algokit/cli/tasks/send_transaction.py:109]: If the transaction group doesn't match on all of the transactions, what would happen?

**Review by robdmoore — APPROVED** [review]

File: src/algokit/cli/tasks/send_transaction.py (hunk 1 above)
Comment (robdmoore) [reply src/algokit/cli/tasks/send_transaction.py:56]: Should we use `transaction` instead to be clearer? What format is it in?

Comment (aorumbayev) [reply src/algokit/cli/tasks/send_transaction.py:56]: @robdmoore renamed into `transaction_id` its just a string value.

File: src/algokit/cli/tasks/send_transaction.py (hunk 2 above)
Comment (aorumbayev) [reply src/algokit/cli/tasks/send_transaction.py:109]: @robdmoore algosdk will handle that and throw `inconsistent group values` error during sending the transaction group

**Pull request merged**